src/main/java/wooteco/subway/dao/LineDao.java

```diff
@@ -0,0 +1,160 @@
+
+        Long lineId = insertAction.executeAndReturnKey(params).longValue();
+        return new Line(lineId, line.getName(), line.getColor());
+    }
+
+    public Line findById(Long id) {
+
+        String sql = "select L.id as line_id, L.name as line_name, L.color as line_color, " +
+            "S.id as section_id, S.distance as section_distance, " +
+            "UST.id as up_station_id, UST.name as up_station_name, " +
+            "DST.id as down_station_id, DST.name as down_station_name " +
+            "from LINE L \n" +
+            "left outer join SECTION S on L.id = S.line_id " +
+            "left outer join STATION UST on S.up_station_id = UST.id " +
+            "left outer join STATION DST on S.down_station_id = DST.id " +
+            "WHERE L.id = :id";
+        Map<String, Long> params = Collections.singletonMap("id", id);
+        List<LineAndSection> result = namedParameterJdbcTemplate.query(sql, params, (rs, rowNum) ->
+            new LineAndSection(
+                rs.getLong("line_id"),
+                rs.getString("line_name"),
+                rs.getString("line_color"),
+                new Section(
+                    rs.getLong("section_id"),
+                    new Station(rs.getLong("up_station_id"), rs.getString("up_station_name")),
+                    new Station(rs.getLong("down_station_id"), rs.getString("down_station_name")),
+                    rs.getInt("section_distance")
+                )
+            ));
+        return mapLineFromLineAndSection(result);
+    }
+
+    public void update(Line newLine) {
+        String sql = "update LINE set name = :name, color = :color where id = :id";
+        SqlParameterSource params = new BeanPropertySqlParameterSource(newLine);
+        namedParameterJdbcTemplate.update(sql, params);
+    }
+
+    public List<Line> findAll() {
+        String sql = "select L.id as line_id, L.name as line_name, L.color as line_color, " +
+            "S.id as section_id, S.distance as section_distance, " +
+            "UST.id as up_station_id, UST.name as up_station_name, " +
+            "DST.id as down_station_id, DST.name as down_station_name " +
+            "from LINE L \n" +
+            "left outer join SECTION S on L.id = S.line_id " +
+            "left outer join STATION UST on S.up_station_id = UST.id " +
+            "left outer join STATION DST on S.down_station_id = DST.id ";
+        List<Map<String, Object>> result = namedParameterJdbcTemplate
+            .queryForList(sql, Collections.emptyMap());
+        Map<Long, List<Map<String, Object>>> resultByLine = result.stream()
+            .collect(Collectors.groupingBy(it -> (Long) it.get("line_id")));
+        return resultByLine.entrySet().stream()
+            .map(it -> mapLine(it.getValue()))
+            .collect(Collectors.toList());
+    }
+
+    private Line mapLine(List<Map<String, Object>> result) {
+        if (result.size() == 0) {
+            throw new RuntimeException();
+        }
```

Comment:
```suggestion
        if (result.isEmpty()) {
            throw new RuntimeException();
        }
```
이부분은 런타임익셉션을 그대로 두신 이유가 있나요?

Reply:
앗🙄🙄 일관되게 커스텀한 Exception으로 다시 전부 수정하겠습니다!!